### PR TITLE
Score

### PR DESCRIPTION
**Pull request by Bickor**
Created Score client and score testing class.

### CODE REVIEW

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        // Calculate score
+        for (RestaurantPageViews restaurant: allPageViews) {
+            double restaurantAverage = calculateRestaurantAveragePageViews(restaurant);
+
+            // Since the data is sorted we can just get the latest pageView for that restaurant.
+            int lastPageViewIndex = restaurant.getPageViews().size() - 1;
+            int latestPageView = restaurant.getPageViews().get(lastPageViewIndex).getCount();
+
+            double score = calculateRawScore(latestPageView, restaurantAverage, systemAverage);
+            restaurantIds.add(Long.parseLong(restaurant.getName()));
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:42]: This can be awfully confusing for someone to use. Consider changing the `RestaurantPageViews` class to have ID, name, and views fields instead. That way, the interface is a lot less confusing to use.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        List<RestaurantPageViews> allPageViews = metricsClient.getAllPageViews("restaurantKey");
+
+        double systemAverage = getSystemAverage(allPageViews);
+
+        // Calculate score
+        for (RestaurantPageViews restaurant: allPageViews) {
+            double restaurantAverage = calculateRestaurantAveragePageViews(restaurant);
+
+            // Since the data is sorted we can just get the latest pageView for that restaurant.
+            int lastPageViewIndex = restaurant.getPageViews().size() - 1;
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:38]: Would it make more sense to call getPageViews() only once instead?

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+     * @return double representing the average pageViews for the system.
+     */
+    private double getSystemAverage(List<RestaurantPageViews> allPageViews) {
+        double sum = 0;
+        double count = 0;
+
+        for (RestaurantPageViews restaurant : allPageViews) {
+            for (int i = 0; i < restaurant.getPageViews().size(); i++) {
+                sum += restaurant.getPageViews().get(i).getCount();
+                count++;
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:70]: Nit: you could probably just update `count` once per restaurant by using `restaurant.getPageViews().size()`. Also, wouldn't it make more sense to put the returned value `restaurant.getPageViews()` into a variable so that you're not constantly calling the `getPageViews()` function?

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+     * @param restaurant has all the pageViews for a specific restaurant.
+     * @return double representing the average pageViews for that restaurant.
+     */
+    private double calculateRestaurantAveragePageViews(RestaurantPageViews restaurant) {
+        double sum = 0;
+        double count = 0;
+
+        for (int i = 0; i < restaurant.getPageViews().size(); i++) {
+            sum += restaurant.getPageViews().get(i).getCount();
+            count++;
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:88]: Nit: you could probably just store the value of `restaurant.getPageViews().size()` in a variable and use that as `count` instead.

File: capstone/src/main/java/com/google/step/servlets/PageViewServlet.java [hunk 5]
```diff
@@ -54,7 +54,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
         String json;
         if (restaurantKey == null && year <= 0) {
-            List<RestaurantPageViews> restaurantPageViewsList = metricsClient.getAllPageViews();
+            List<RestaurantPageViews> restaurantPageViewsList = metricsClient.getAllPageViews("restaurantName");
             json = gson.toJson(restaurantPageViewsList);
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/servlets/PageViewServlet.java:58]: If you change the data model of `RestaurantPageViews` and you don't want to send the ID field to the front end, you might have to change this.

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java [hunk 6]
```diff
@@ -116,7 +116,7 @@ public WeeklyPageView getCurrentPageViews(String restaurantKey) {
      * @return returns List<RestaurantPageViews> with the PageView list inside each instance
      *         of RestaurantPageViews sorted by week and year in ascending order.
      */
-    public List<RestaurantPageViews> getAllPageViews() {
+    public List<RestaurantPageViews> getAllPageViews(String identificationType) {
```
Comment (kathleen-lund) [inline capstone/src/main/java/com/google/step/clients/MetricsClient.java:119]: Why did you add this parameter?

Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/MetricsClient.java:119]: I added the parameter because Josh needed to have the restaurantKey instead of the restaurantName for the scoring stuff. That way I could choose which one I got depending on the need.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 2 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:38]: In this instance it is being called only once but I'll change it on the other places.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 3 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:70]: Yup done.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 4 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:88]: Done

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 1 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:42]: We talked about it over gvc and fixed it.

File: capstone/src/main/java/com/google/step/servlets/PageViewServlet.java (hunk 5 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/servlets/PageViewServlet.java:58]: We talked about it about it over gvc.

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java (hunk 6 above)
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/clients/MetricsClient.java:119]: Please update the javadoc to explain what identificationType is.

What are all the identification types? Also, should this live in an enum?

File: capstone/src/main/java/com/google/step/servlets/PageViewServlet.java [hunk 7]
```diff
@@ -54,7 +54,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
         String json;
         if (restaurantKey == null && year <= 0) {
-            List<RestaurantPageViews> restaurantPageViewsList = metricsClient.getAllPageViews();
+            List<RestaurantPageViews> restaurantPageViewsList = metricsClient.getAllPageViews("restaurantName");
```
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/servlets/PageViewServlet.java:57]: hmmm, to help decouple things, I do think this sort of thing really should be an enum. Then, you don't have to require callers of your client to know the property name on an entity.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+import java.util.HashMap;
+import java.util.Collections;
+
+
+/**
+ * Will take care of calculating the score for each restaurant.
+ */
+public class ScoreClient {
+
+    private final float minimum = 10.0f;
```
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:19]: 1. Minimum what? This isn't very descriptive.
2. Since this is basically a constant, I'd make it `private static final`
3. In general, just don't use `float`. There's almost never a good reason not to make this a `double`.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 2 above)
Comment (tzsebe) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:39]: It's being called twice in this instance too (line 38 and line 39)

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+        double count = 0;
+
+        for (RestaurantPageViews restaurant : allPageViews) {
+            for (int i = 0; i < restaurant.getPageViews().size(); i++) {
+                sum += restaurant.getPageViews().get(i).getCount();
+                count++;
+            }
+        }
+
+        return sum / count;
```
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:76]: Can count be zero?

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 1 above)
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:43]: I thought the restaurant name was a string, and the restaurant key was the Long.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+public class ScoreClient {
+
+    private final float minimum = 10.0f;
+    private final MetricsClient metricsClient = new MetricsClient();
+
+    /**
+     * Only method to be called and calculates the scores of each restaurant.
+     * @return an unmodifiable list of restaurantScore class.
+     */
+    public List<RestaurantScore> calculateScores() {
```
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:26]: General comment:

This is something you can do AFTER the presentation if you have time, so don't feel the need to refactor now.

It would likely be a lot more efficient (and possibly easier to read) if you explicitly extracted things into arrays, did the math, and then recombined, as follows:

(forgive broken pseudocode, it's just to give you an idea of the general pattern)

```
// Unpack
Map<Long, Integer> indexes = new HashMap<>();
double scores[] = new double[allPageViews.size()];
int pageViews[] = new int[allPageViews.size()];
int idx = 0;
for (RestaurantPageViews pageView : allPageViews) {
    indexes.add(pageView.getRestaurantKey(), idx);
    scores[idx] = 0.0; // or whatever you need it to be
    pageViews[idx] = pageView.getCount(); // or whatever it is...
    ++idx;  
}

// Run your scoring algorithm on the arrays; you don't need to keep looking up restaurant keys/ids/names, they don't matter for this part.
(...)

// Pack your results as needed.
for (Map::Entity<Long, Integer> idx : indexes) {
    result.add(<something something> idx.Key(), <something something> scores[idx.Value()]);
}
return result;
```

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java (hunk 6 above)
Comment (mkmg) [reply capstone/src/main/java/com/google/step/clients/MetricsClient.java:119]: I'd prefer we add an additional field for the restaurant key to the RestaurantPageViews for the following reason:
1. The documentation and function/field names of the RestaurantPageViews class do not work well with this set-up. For example, will Josh need to call getName() to get the restaurant key? I think it would be easier to use and understand if there was a new restaurantKey field and getRestaurantKey method.
2. We already have the entity, so there's no significant performance cost to populating the RestauarantPageViews object with both values (e.g. no need for an extra query).

Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/MetricsClient.java:119]: Fixed javadoc.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+/**
+ * Will take care of calculating the score for each restaurant.
+ */
+public class ScoreClient {
+
+    private final float minimum = 10.0f;
+    private final MetricsClient metricsClient = new MetricsClient();
+
+    /**
+     * Only method to be called and calculates the scores of each restaurant.
```
Comment (mkmg) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:23]: Nit: Consider removing "Only method to be called and". Users should be able to tell this by looking for all the public functions. Also this way if you add another public function you don't need to change this comment.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+     * @return an unmodifiable list of restaurantScore class.
+     */
+    public List<RestaurantScore> calculateScores() {
+        Map<Long, Double> scoreMap = new HashMap<>();
```
Comment (mkmg) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:27]: Why is this a Map<Long, Double> instead of a list?

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 8 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:19]: Changed!

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 2 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:39]: Ah true. Changed.

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java (hunk 6 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/MetricsClient.java:119]: Yeah that does make sense! Changed!

File: capstone/src/main/java/com/google/step/servlets/PageViewServlet.java (hunk 7 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/servlets/PageViewServlet.java:57]: Same comment as above.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 1 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:43]: Yup changed. Now the method is for getId()

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 9 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:76]: No its not possible for the count to be zero since in that case the restaurant instance wouldn't even exist. The minimum for count would be 1 in this case.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 11 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:23]: Done!

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 12 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:27]: In order to keep track of the restaurant score at the same time. That way, by also having a list with the id's I can iterate easier through the map when standardizing and normalizing the score.

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java [hunk 13]
```diff
@@ -140,13 +142,14 @@ public WeeklyPageView getCurrentPageViews(String restaurantKey) {
                 List<WeeklyPageView> currRestaurantList = new ArrayList<>();
                 currRestaurantList.add(new WeeklyPageView(week, year, count));
                 restaurantNameMap.put(name, currRestaurantList);
```
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/clients/MetricsClient.java:144]: Is it safe to assume that restaurant name is unique?

I'm okay with it for demo purposes, if you want to back-burner it.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 14]
```diff
@@ -15,18 +16,18 @@
  */
 public class ScoreClient {
 
-    private final float minimum = 10.0f;
+    private final double minimumWeeks = 10.0;
```
Comment (tzsebe) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:19]: Make it static

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 12 above)
Comment (mkmg) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:27]: I may be missing it, but I don't don't see any place where you use the map to get a score for a specific key (other than iterating over every key in the map). 

If you only ever iterate over every restaurant/score pair, then this could be a list (or even a set) of items (you could create a class to store the long and double values). A list would be more space efficient (under the hood the map uses an array whose length is greater than the number of keys stored in the map). 
 
Even if you decide to stay with the map, there is no need to have a separate list with the ids. You can iterate over the map using the entry set method and a for each loop: https://www.geeksforgeeks.org/iterate-map-java/

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java [hunk 15]
```diff
@@ -128,9 +128,11 @@ public WeeklyPageView getCurrentPageViews(String restaurantKey) {
         // Map to maintain a reference to each restaurantList and make searching for them easy.
         // The map and the list share the reference to each restaurantPageViews.
         Map<String, List<WeeklyPageView>> restaurantNameMap = new HashMap<>();
+        Map<String, String> restaurantIdMap = new HashMap<>();
```
Comment (mkmg) [inline capstone/src/main/java/com/google/step/clients/MetricsClient.java:131]: Nit: Consider making a private inner class that stores a restaurant name and a List<WeeklyPageView>.
Then you can have a single map from the restaurant id (which is guaranteed to be unique for each restaurant) to this class.

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java [hunk 16]
```diff
@@ -128,9 +128,11 @@ public WeeklyPageView getCurrentPageViews(String restaurantKey) {
         // Map to maintain a reference to each restaurantList and make searching for them easy.
         // The map and the list share the reference to each restaurantPageViews.
         Map<String, List<WeeklyPageView>> restaurantNameMap = new HashMap<>();
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/clients/MetricsClient.java:130]: If you're doing a dual-map approach, would it make more sense to key the maps by the `restaurantKey` field, since you can guarantee uniqueness of that field for a single restaurant?

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 9 above)
Comment (josh8551021) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:77]: What if the `List<RestaurantPageViews>` input is an empty list?

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+        return score;
+    }
+
+    /**
+     * Standardizes the scores for the entire map of scores using z - score calculations.
+     * @param scoreMap map that holds the restaurant id as a key and the score of that restaurant as a value.
+     * @param restaurantIds list that holds the restaurant ids to get the values of the map.
+     */
+    private void standardizeScores(Map<Long, Double> scoreMap, List<Long> restaurantIds) {
+        double sum = 0.0f;
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:117]: Nit: You probably shouldn't have the `f` at the end of the literal.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+
+        for (Long restaurantId : restaurantIds) {
+            sum += scoreMap.get(restaurantId);
+        }
+
+        double mean = sum / restaurantIds.size();
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:123]: What happens in the case that the `restaurantIds` parameter is epmty?

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+                + (minimumWeeks / (latestPageViews + minimumWeeks)) * systemAverage;
+        return score;
+    }
+
+    /**
+     * Standardizes the scores for the entire map of scores using z - score calculations.
+     * @param scoreMap map that holds the restaurant id as a key and the score of that restaurant as a value.
+     * @param restaurantIds list that holds the restaurant ids to get the values of the map.
+     */
+    private void standardizeScores(Map<Long, Double> scoreMap, List<Long> restaurantIds) {
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:116]: Is there ever a case that `restaurantIds` has an element that isn't a key of the score map, or vice versa? It might be worth using a single input in this case, and iterating over the entry set of the map instead.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+        }
+    }
+
+    /**
+     * Normalizes the scores into a range between 0 and 1 using min max normalizing.
+     * It also inverts the score so a 1 is a low score and a 0 is a high score.
+     * @param scoreMap map that holds the restaurant id as a key and the score of that restaurant as a value.
+     * @param restaurantIds list that holds the restaurant ids to get the values of the map.
+     */
+    private void normalizeScores(Map<Long, Double> scoreMap, List<Long> restaurantIds) {
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:145]: Same here about the second parameter.

File: capstone/src/main/java/com/google/step/data/RestaurantPageViews.java [hunk 21]
```diff
@@ -33,6 +35,12 @@ public String getName() {
         return pageViews;
     }
 
+    /**
+     * Gets the id of the instance
+     * @return String representing the id of the restaurant.
+     */
+    public String getId() { return id; }
```
Comment (josh8551021) [inline capstone/src/main/java/com/google/step/data/RestaurantPageViews.java:42]: Nit: it doesn't matter that much, but consider changing this to be consistent with the other getters.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java [hunk 22]
```diff
@@ -0,0 +1,164 @@
...
+        for (RestaurantPageViews restaurant: allPageViews) {
+            double restaurantAverage = calculateRestaurantAveragePageViews(restaurant);
+
+            // Since the data is sorted we can just get the latest pageView for that restaurant.
+            List<WeeklyPageView> pageViewList = restaurant.getPageViews();
+            int lastPageViewIndex = pageViewList.size() - 1;
+            int latestPageView = pageViewList.get(lastPageViewIndex).getCount();
+
+            double score = calculateRawScore(latestPageView, restaurantAverage, systemAverage);
+            restaurantIds.add(Long.parseLong(restaurant.getId()));
```
Comment (kathleen-lund) [inline capstone/src/main/java/com/google/step/clients/ScoreClient.java:44]: Does the `RestaurantPageViews` class have a `getId()` method?

Comment (josh8551021) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:44]: I believe this field was added in his most recent iteration.

Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:44]: Yup!

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java (hunk 16 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/MetricsClient.java:130]: The Id should also be unique.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 9 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:77]: Fixed in the only public method.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 10 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:26]: I did a similar method to what you explained and didn't even need to use maps! Fixed!

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 12 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:27]: I basically changed everything to use lists and should be a lot quicker now! I'm no longer using maps at all.

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java (hunk 13 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/MetricsClient.java:144]: Fixed so it utilized id instead which is unique per restaurant.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 14 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:19]: Changed to be a local vairable.

File: capstone/src/main/java/com/google/step/clients/MetricsClient.java (hunk 15 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/MetricsClient.java:131]: Done! It looks a lot better and less confusing! Thanks for the tip!

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 17 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:117]: Yup fixed!

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 18 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:123]: It won't be empty since I added a check in the public method to return if there are no elements when getting all page views.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 19 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:116]: Changed so there is no need for a map at all.

File: capstone/src/main/java/com/google/step/clients/ScoreClient.java (hunk 20 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/clients/ScoreClient.java:145]: same comment as above.

File: capstone/src/main/java/com/google/step/data/RestaurantPageViews.java (hunk 21 above)
Comment (Bickor) [reply capstone/src/main/java/com/google/step/data/RestaurantPageViews.java:42]: Done.
